cmd/flags/auth.go

```diff
@@ -178,3 +211,42 @@ func issuerCertificates(vipr *viper.Viper) ([]*x509.Certificate, error) {
 
 	return certs, nil
 }
+
+func apiKeyCsvFile(vipr *viper.Viper) (map[string]*apikey.UserNameAndGroups, error) {
+	// Open the file
+	csvFileName := vipr.GetString(authAPIKeyFileViperKey)
+	csvfile, err := os.Open(csvFileName)
+	if err != nil {
+		return nil, err
+	}
+	defer func(csvfile *os.File) {
+		_ = csvfile.Close()
+	}(csvfile)
+
+	// Parse the file
+	r := csv.NewReader(csvfile)
+	// Set separator as ";"
+	r.Comma = ';'
+	// ignore comments in file
+	r.Comment = '#'
+
+	retFile := make(map[string]*apikey.UserNameAndGroups)
+
+	// Iterate through the lines
+	for {
+		// Read each line from csv
+		cells, err := r.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		retFile[cells[0]] = &apikey.UserNameAndGroups{UserName: cells[1],
```

Comment:
I am missing the entire error handling here, we would panic in case csv is not formatted as expected, instead we should manage that and return to the user a proper error message

Reply:
Yes that part is missing, I will had it

Reply:
I would rather return a nil config instead of panicing